### PR TITLE
(PUP-7117) Updated acceptance test for utf-8 characters in file resource

### PR DESCRIPTION
**Pull request by seangriff**
The original version of this test was modified to set the environment
variable LANG to "en_US.UTF-8" on each call to on() and apply_manifest_on().
There are some VM pooler templates that do not set this by default.

### CODE REVIEW

**Comment by MosesMendoza** [comment]
@seangriff can you post link to the output of a successful ad-hoc run before we merge this?

**Comment by seangriff** [comment]
@MosesMendoza

successful output of ad-hoc run:

https://gist.github.com/seangriff/ff53e8b17ac82b8605b0b0bd18f327d3

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        }
+      )
+      result = on(
+        agent, "cat #{agent_file}", :enviornment => {:LANG => "en_US.UTF-8"}
+      )
+      assert_equal(result.exit_code, 0)
+      assert_match(
+        /#{utf8chars}/,
+        result.stdout,
+        "result stdout did not contain"
```
Comment (MosesMendoza) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: I think this output message may have gotten truncated somewhere? was it intended to include the utf 8 chars?

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        agent,
+        manifest,
+        {
+          :acceptable_exit_codes => (0..2),
+          :catch_failures => true, 
+          :environment => {:LANG => "en_US.UTF-8"}
+        }
+      )
+      result = on(
+        agent, "cat #{agent_file}", :enviornment => {:LANG => "en_US.UTF-8"}
```
Comment (MosesMendoza) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: spelling s/enviornment/environment

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        result.stdout,
+        "result stdout did not contain"
+      )
+    end
+
+    step "Drift correction" do
+      on(
+        agent,
+        "echo '' > #{agent_file}",
+        :enviornment => {:LANG => "en_US.UTF-8"}
```
Comment (MosesMendoza) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: spelling s/enviornment/environment

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      )
+      result = on(
+        agent,
+        "cat #{agent_file}",
+        :enviornment => {:LANG => "en_US.UTF-8"}
```
Comment (MosesMendoza) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: spelling s/enviornment/environment

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      )
+      assert_equal(result.stdout.chomp, "", "expected empty file")
```
Comment (MosesMendoza) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: I'm not sure this assertion / tests is necessary? On line 54 the file is effectively zeroed with `echo '' > file` and these `on agent` runs add a lot of time to tests.

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      )
+      result = on(
+        agent,
+        "cat #{agent_file}",
+        :environment => {:LANG => "en_US.UTF-8"}
+      )
+      assert_match(
+        /#{utf8chars}/,
+        result.stdout,
+        "result stdout did not contain"
```
Comment (MosesMendoza) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: same question as above regarding output

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 7]
```diff
@@ -0,0 +1,85 @@
+test_name 'utf-8 characters in resource title and param values' do
+
+  confine :except, :platform => [
+    'windows',    # PUP-6983
+    'eos-4',      # PUP-7146
+    'cumulus',    # PUP-7147
+    'cisco_ios',  # PUP-7150
```
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: Only cisco_ios.  This test works when run on cisco-nexus (template: cisco-exr-9k-x86_64).

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  ]   
+
+  # utf8chars = "€‰ㄘ万竹ÜÖ"
+  utf8chars = "\u20ac\u2030\u3118\u4e07\u7af9\u00dc\u00d6"
+  agents.each do |agent|
+    agent_file = agent.tmpfile("file" + utf8chars) 
+    # remove this file, so puppet can create it and not merely correct
+    # its drift.
+    on(agent, "rm -rf #{agent_file}", :enviornment => {:LANG => "en_US.UTF-8"})
```
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: fixed. Thx.

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+      result = apply_manifest_on(
+        agent,
+        manifest,
+        {
+          :acceptable_exit_codes => (0..2),
+          :catch_failures => true, 
+          :environment => {:LANG => "en_US.UTF-8"}
+        }
+      )
+      result = on(
```
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: fixed. Thx.

Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: Changed on() invocation to use the block format as suggested.

File: acceptance/tests/utf8/utf8-in-file-resource.rb (hunk 2 above)
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: fixed. Thx.

File: acceptance/tests/utf8/utf8-in-file-resource.rb [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+        agent, "cat #{agent_file}", :enviornment => {:LANG => "en_US.UTF-8"}
+      )
+      assert_equal(result.exit_code, 0)
```
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: Correct.  Dropped assert_equal() test for exit_code.

File: acceptance/tests/utf8/utf8-in-file-resource.rb (hunk 1 above)
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: fixed. Thx.

File: acceptance/tests/utf8/utf8-in-file-resource.rb (hunk 3 above)
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: fixed. Thx.

File: acceptance/tests/utf8/utf8-in-file-resource.rb (hunk 4 above)
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: fixed. Thx.

File: acceptance/tests/utf8/utf8-in-file-resource.rb (hunk 5 above)
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: Concurred.  Assertion removed.

File: acceptance/tests/utf8/utf8-in-file-resource.rb (hunk 6 above)
Comment (seangriff) [inline acceptance/tests/utf8/utf8-in-file-resource.rb]: fixed. Thx.

**Pull request merged**